### ISSUE TITLE
env not working inside .babelrc files

### ISSUE THREAD

**Issue by R4VANG3R** [nrwl/nx#3351]
## Current Behavior

.babelrc  
```
{
  "presets": ["@nrwl/react/babel"],
  "env": {
    "development": {
      "plugins": [
        [
          "styled-components",
          {
            "pure": true,
            "ssr": true,
            "displayName": true
          }
        ]
      ]
    },
    "production": {
      "plugins": [
        [
          "styled-components",
          {
            "pure": true,
            "ssr": true,
            "displayName": false,
            "fileName": false
          }
        ]
      ]
    }
  }
}
```
With this config it will always disable the displayNames, it seems to only react to the config if you put the `plugins` outside of `env`.

## Expected Behavior

I expect the configs in `env` to be picked up based on the current `NODE_ENV`.

## Steps to Reproduce

Example repo: https://github.com/R4VANG3R/nx-examples
- Run the cart app
- The div with "Foo" in it is styled and has an unreadable name.

### Environment

@nrwl/angular : Not Found
  @nrwl/cli : 9.5.1
  @nrwl/cypress : 9.5.1
  @nrwl/eslint-plugin-nx : 9.5.1
  @nrwl/express : 9.5.1
  @nrwl/jest : 9.5.1
  @nrwl/linter : 9.5.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : 9.5.1
  @nrwl/react : 9.5.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 9.5.1
  @nrwl/web : 9.5.1
  @nrwl/workspace : 9.5.1
  typescript : 3.9.6

**Comment by vaunus** [comment]
I am seeing an issue caused by this commit that is preventing Babel from working correctly within my customised NX configuration.

I am trying to add ``` "import/resolver": { "babel-module": {} } ``` to my `.eslintrc` as I would like to add some JavaScript-only module aliases. I am using `babel-eslint` as my `.js` file linter but am seeing linting errors as Babel fails to resolve the module. 

As soon as I add the above to ESLint I get the following Babel error:
```
[Function: caller]
[eslint-import-resolver-babel-module] TypeError: [BABEL] /nxtest/apps/japp/src/app/app.js: 

Cannot read property 'isModern' of undefined (While processing: "/nxtest/node_modules/@nrwl/web/babel.js")
```

It seems that the line causing this is [packages/web/babel.ts#L8](https://github.com/nrwl/nx/blob/master/packages/web/babel.ts#L8) as `caller` is always `undefined` and that `isModern` property is being set by a custom `web-babel-loader` introduced in this commit.

Is there a recommended way I can get the above working or should the line above be modified to something like:
`const isModern = api.caller((caller) => caller && caller.isModern);`?

**Comment by jordandlaman** [comment]
Has anyone actually confirmed this is fixed?  My team is on 10.1.0 and while it appears the fix is in -- the env option in either the root babel.config.json or the apps .babelrc does not seem to be respected at all.

**Comment by jaysoo** [comment]
@vaunus We'll provide a patch in 10.2 for this bug, the `isModern` flag should be marked as optional and defaults to true. It's purpose is only for differential loading (if you opt into IE11 support).

@jordandlaman what are you trying to override with the env option? We have a e2e test to check for dev vs prod builds using different styled-component options. https://github.com/nrwl/nx/blob/master/e2e/react/src/react.test.ts#L159-L183

**Comment by jordandlaman** [comment]
> @jordandlaman what are you trying to override with the env option? We have a e2e test to check for dev vs prod builds using different styled-component options. https://github.com/nrwl/nx/blob/master/e2e/react/src/react.test.ts#L159-L183

Was attempting to get the @cypress/code-coverage module working with our react project.  They recommend using the istanbul plugin in the babelrc file to instrument the code.  If slotted into the plugins section it works fine but obviously we don't it instrumenting the code during the build so I hoped to take advantage of the env option in babel to set NODE_ENV=test and ensure that instrumenting is only done during e2e runs or ci stages.

I have tried variations of something like

```
{
  "presets": ["@nrwl/web/babel"],
  "babelrcRoots": ["*"],
  "env": {
    "test": {
      "plugins": ["istanbul"]
    }
  },
  "plugins": ["lodash"]
}
```
But no matter how I set that env or what I name it didn't seem to make a difference.  Istanbul never instruments the code.  If were to swap istanbul into the vanilla plugins section it instruments everything.  Currently I using an if statement in the app .babelrc.js file to conditionally swap in istanbul but thats hit a road block as it is only instrumenting a few files if the plugin is called from the app babel config and not the root babel config.  Changing the babel.config.json to a js file has proven equally as difficult as it acts like its not using the correct typescript compilation settings.  Errors like 
```
"Unexpected reserved word 'interface'"
```
and 
```
" Support for the experimental syntax 'flow' isn't currently enabled "
```
So if there is something obvious I am doing wrong like is env not able to use anything but the vars of production and development?
